rs/replicated_state/BUILD.bazel

```diff
@@ -35,6 +35,7 @@ DEPENDENCIES = [
     "@crate_index//:maplit",
     "@crate_index//:nix",
     "@crate_index//:prometheus",
+    "@crate_index//:prost",
```

Comment:
Not sure I see why prost moves from dev dependencies to actual dependencies? Isn't the encoding done within a debug assert?

Reply:
It is, and I tried to only import `prost` as a test dependency. But `dev-dependencies` applies to `#[cfg(test)]` (i.e. when building tests) and `debug_assert!()` is gated on `#[cfg(debug_assertions)]` (i.e. building without optimizations). And the two are unfortunately (for this use case) independent.

We depend on `prost` indirectly anyway, so this doesn't really add any overhead.

(I guess I could put the `debug_assert!()` along with `use prost` behind a `#[cfg(test)]`. Then it could remain a dev dependency.)

Reply:
I see. Thanks for the explanation.

Reply:
OTOH, that last approach may not be such a good idea. The reason why this failed in Mathias' test (which is clearly not a unit test in `ic_replicated_state`) is precisely because it's a `debug_assert!()` that gets compiled into any and all tests.

If we were to put it behind a `#[cfg(test)]` it would only be compiled into `ic_replicated_state` unit tests, so it would be a lot less likely to catch any actual issue.

Reply:
Oh... seems I somehow overlooked the suggestion for the approach that would keep it as a dev dependency and assumed that we would leave things as is when closing this thread. 

This is indeed a good reason for keeping prost as an actual dependency.